projects/igniteui-angular/src/lib/date-range-picker/date-range-picker.component.ts

```diff
@@ -790,46 +798,57 @@ export class IgxDateRangePickerComponent extends DisplayDensityBase
         });
     }
 
+    private parseMinMax(value: string | Date, isStart = true): Date | null {
+        let minMax = DatePickerUtil.parseDate(value);
+        let startEnd: IgxDateRangeStartComponent | IgxDateRangeEndComponent;
+        if (!minMax && this.hasProjectedInputs) {
+            startEnd = this.projectedInputs
+                .filter(i => i instanceof (isStart
+                    ? IgxDateRangeStartComponent
+                    : IgxDateRangeEndComponent))[0];
+            minMax = isStart
+                ? DatePickerUtil.parseDate(startEnd.dateTimeEditor.minValue)
+                : DatePickerUtil.parseDate(startEnd.dateTimeEditor.maxValue);
+        }
+
+        return minMax;
+    }
+
     private updateCalendar(): void {
         this.calendar.disabledDates = [];
-        let minValue: Date = DatePickerUtil.parseDate(this.minValue);
-        if (!minValue && this.hasProjectedInputs) {
-            const start = this.projectedInputs.filter(i => i instanceof IgxDateRangeStartComponent)[0];
-            if (start) {
-                minValue = DatePickerUtil.parseDate(start.dateTimeEditor.minValue);
-            }
-        }
+        const minValue = this.parseMinMax(this.minValue);
         if (minValue) {
             this.calendar.disabledDates.push({ type: DateRangeType.Before, dateRange: [minValue] });
         }
-
-        let maxValue: Date = DatePickerUtil.parseDate(this.maxValue);
-        if (!maxValue && this.hasProjectedInputs) {
-            const end = this.projectedInputs.filter(i => i instanceof IgxDateRangeEndComponent)[0];
-            if (end) {
-                maxValue = DatePickerUtil.parseDate(end.dateTimeEditor.maxValue);
-            }
-        }
+        const maxValue = this.parseMinMax(this.maxValue, false);
         if (maxValue) {
             this.calendar.disabledDates.push({ type: DateRangeType.After, dateRange: [maxValue] });
         }
 
         const range: Date[] = [];
-        if (this.value) {
-            if (this.value.start) {
-                range.push(this.value.start);
-            }
-            if (this.value.end) {
-                range.push(this.value.end);
-            }
+        if (this.value
+            && this.value.start && this.value.end
+            && this.valueInRange(this.value, minValue, maxValue)) {
+            range.push(this.value.start, this.value.end);
         }
 
         if (range.length > 0) {
             this.calendar.selectDate(range);
-            this.calendar.viewDate = range[0];
         } else {
             this.calendar.deselectDate();
         }
+        this.calendar.viewDate = range[0] || new Date();
+    }
+
+    private valueInRange(value: DateRange, minValue?: Date, maxValue?: Date): boolean {
```

Comment:
if for some unknown reason we have `value.start` later than `value.end` this will not work